src/WebJobs.Script.WebHost/Diagnostics/HealthChecks/HealthCheckWaitMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.HealthChecks
+{
+    public class HealthCheckWaitMiddleware(RequestDelegate next, IScriptHostManager manager)
```

Comment:
Where and how are we hooking this up to the request processing pipeline?  I did not see this middleware registered anywhere.

Reply:
It is not hooked up yet, will be doing that in a later PR. I am 'staging' all the different parts of health checks first, then integrating them into startup as the last step.